doc/news/changes/incompatibilities/20250116Bangerth

```diff
@@ -0,0 +1,22 @@
+Changed: The header file `deal.II/grid/tria.h` used to automatically
+include the header file `deal.II/grid/tria_description.h` that
+declares classes such as CellData and SubCellData. But this led to a
+circular set of inclusions because `deal.II/grid/tria_description.h`
+also included `deal.II/grid/tria.h`, something that C++ allows but
+that leads to other problems. As a consequence, `deal.II/grid/tria.h`
+now no longer includes `deal.II/grid/tria_description.h`. If your
+program uses the CellData, SubCellData, or similar classes previously
+declared in `deal.II/grid/tria_description.h`, then you may want to
+explicitly include `deal.II/grid/tria_cell_data.h` (for the
```

Comment:
```suggestion
explicitly include `deal.II/grid/cell_data.h` (for the
```

Reply:
In order to make progress with the cycles in #17992, and because I've burned enough electricity with this PR already, let me merge as-is and fix the file name in a separate PR that also fixes another misspelling.